plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -1782,4 +1783,193 @@ describe('PersonState.update()', () => {
             )
         })
     })
+    describe('on persons merges', () => {
+        it('postgres and clickhouse get updated', async () => {
+            const first: Person = await hub.db.createPerson(
+                timestamp,
+                {},
+                {},
+                {},
+                teamId,
+                null,
+                false,
+                uuid.toString(),
+                ['first']
+            )
+            const second: Person = await hub.db.createPerson(
+                timestamp,
+                {},
+                {},
+                {},
+                teamId,
+                null,
+                false,
+                uuid2.toString(),
+                ['second']
+            )
+
+            const state: PersonState = personState({}, first)
+            jest.spyOn(hub.db.kafkaProducer, 'queueMessages')
+            jest.spyOn(state, 'aliasDeprecated').mockImplementation()
+            await state.mergePeople({
+                mergeInto: first,
+                mergeIntoDistinctId: 'first',
+                otherPerson: second,
+                otherPersonDistinctId: 'second',
+                timestamp: timestamp,
+                totalMergeAttempts: 0,
+            })
+            await hub.db.kafkaProducer.flush()
+
+            expect(hub.db.updatePersonDeprecated).toHaveBeenCalledTimes(1)
+            expect(state.aliasDeprecated).not.toHaveBeenCalled()
+            expect(hub.db.kafkaProducer.queueMessages).toHaveBeenCalledTimes(1)
+            // verify Postgres persons
+            const persons = await hub.db.fetchPersons()
+            expect(persons.length).toEqual(1)
+            expect(persons[0]).toEqual(
+                expect.objectContaining({
+                    id: expect.any(Number),
+                    uuid: uuid.toString(),
+                    properties: {},
+                    created_at: timestamp,
+                    version: 1,
+                    is_identified: true,
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['first', 'second']))
+
+            // verify ClickHouse persons
+            await delayUntilEventIngested(() => fetchPersonsRowsWithVersionHigerEqualThan(), 2) // wait until merge and delete processed
+            const clickhousePersons = await fetchPersonsRows() // but verify full state
+            expect(clickhousePersons.length).toEqual(2)
```

Comment:
I would remove this check, we can assume that this will be met if the below expect passes, and we'll likely get a better error output from it.